qiskit_ibm_runtime/runtime_job.py

```diff
@@ -319,13 +323,40 @@ def _set_error_message(self, job_response: Dict) -> None:
             job_response: Job response from runtime API.
         """
         if self._status == JobStatus.ERROR:
-            job_result_raw = self._api_client.job_results(job_id=self.job_id)
-            self._error_message = API_TO_JOB_ERROR_MESSAGE[
-                job_response["status"].upper()
-            ].format(self.job_id, job_result_raw)
+            self._error_message = self._error_msg_from_job_response(job_response)
         else:
             self._error_message = None
 
+    def _error_msg_from_job_response(self, response: Dict) -> str:
+        """Returns the error message from an API response
+
+        Args:
+            response: Job response from the runtime API.
+
+        Returns:
+            Error message.
+        """
+        status = response["state"]["status"].upper()
+        job_result_raw = self._api_client.job_results(job_id=self.job_id)
+        error_msg = API_TO_JOB_ERROR_MESSAGE["FAILED"]
+        if status == "CANCELLED" and self._reason == "RAN TOO LONG":
+            error_msg = API_TO_JOB_ERROR_MESSAGE["CANCELLED - RAN TOO LONG"]
+        return error_msg.format(self.job_id, job_result_raw)
+
+    def _status_from_job_response(self, response: Dict) -> str:
+        """Returns the job status from an API response
+
+        Args:
+            response: Job response from the runtime API.
+
+        Returns:
+            Job status.
+        """
+        mapped_job_status = API_TO_JOB_STATUS[response["state"]["status"].upper()]
+        if mapped_job_status == JobStatus.CANCELLED and self._reason == "RAN TOO LONG":
+            mapped_job_status = JobStatus.ERROR
```

Comment:
Why is cancelled ran too long mapped to error? I think it should be mapped to Cancelled status.

Reply:
Hmm didn't realize that, thanks, fine with me.